spec/engine/miq_ae_method_service/miq_ae_service_methods_spec.rb

```diff
@@ -162,5 +162,32 @@ def category_create_script
       ct.reload
       expect(ct.entries.collect(&:name).include?('fred')).to be_truthy
     end
+
+    context "#create_service_provision_request" do
+      let(:options) { {:fred => :flintstone} }
+      let(:svc_options) { {:dialog_style => "medium"} }
+      let(:user) { FactoryGirl.create(:user_with_group) }
+      let(:template) { FactoryGirl.create(:service_template_ansible_playbook) }
+      let(:miq_request) { FactoryGirl.create(:service_template_provision_request) }
+      let(:svc_template) do
+        MiqAeMethodService::MiqAeServiceServiceTemplate.find(template.id)
+      end
+      let(:workspace) do
+        double("MiqAeEngine::MiqAeWorkspaceRuntime",
+               :root               => options,
+               :persist_state_hash => {},
+               :ae_user            => user)
+      end
+      let(:miq_ae_service) { MiqAeService.new(workspace) }
+
+      it "create service request" do
+        allow(workspace).to receive(:disable_rbac)
+        expect(svc_template).to receive(:instance_variable_get).with('@object').and_return(template)
+        expect(template).to receive(:provision_request).with(user, svc_options).and_return(miq_request)
+
+        result = miq_ae_service.execute(:create_service_provision_reques, svc_template, svc_options)
```

Comment:
@mkanoor, aka. Hawk Eye ;)
I corrected it. It seems the problem comes from the fact that MiqAeMethodService class is not know in this context. I also tried to change the test to:

```ruby
    it "#create_service_provision_request" do
      options = {:fred => :flinstone}
      svc_options = {:dialog_style => "medium"}
      template = FactoryGirl.create(:service_template_ansible_playbook)
      miq_request = FactoryGirl.create(:service_template_provision_request)
      svc_template = MiqAeMethodService::MiqAeServiceServiceTemplate.find(template.id)
      workspace = double("MiqAeEngine::MiqAeWorkspaceRuntime", :root => options, :persist_state_hash => {}, :ae_user => @user)
      #miq_ae_service = MiqAeMethodService.new(workspace)

      allow(workspace).to receive(:disable_rbac)
      expect(svc_template).to receive(:instance_variable_get).with('@object').and_return(template)
      expect(template).to receive(:service_provision_request).with(@user, svc_options).and_return(miq_request)

      result = invoke_ae.execute(:create_service_provision_request, svc_template.inspect, svc_options.inspect)
      expect(result).to be_kind_of(MiqAeMethodService::MiqAeServiceMiqRequest)
    end
```

But invoke_ae doesn't really give ```$evm``` object and tells me

```ruby
undefined method `execute' for #<MiqAeEngine::MiqAeWorkspaceRuntime:0x0000000e3db518>
```